diff -dur jna-4.2.1.orig/native/Makefile jna-4.2.1/native/Makefile
--- jna-4.2.1.orig/native/Makefile	2015-10-18 14:40:13.000000000 +0200
+++ jna-4.2.1/native/Makefile	2015-10-27 12:17:47.000000000 +0100
@@ -69,6 +69,7 @@
 FFI_ENV=CC="$(CC)" CFLAGS="$(COPT) $(CDEBUG)" CPPFLAGS="$(CDEFINES)"
 FFI_CONFIG=--enable-static --disable-shared --with-pic=yes
 endif
+LIBBASE=$(LIBPFX)jnidispatch$(JNISFX)
 LIBRARY=$(BUILD)/$(LIBPFX)jnidispatch$(JNISFX)
 TESTLIB=$(BUILD)/$(LIBPFX)testlib$(LIBSFX)
 TESTLIB_JAR=$(BUILD)/$(LIBPFX)testlib-jar$(LIBSFX)
@@ -260,10 +261,10 @@
 PCFLAGS+=-fPIC
 CDEFINES+=-DHAVE_PROTECTION
 ifeq ($(DYNAMIC_LIBFFI),true)
-LDFLAGS+=-Wl,-soname,$@
+LDFLAGS+=-Wl,-soname,$(LIBBASE)
 else
 # Ensure we bind to local libffi symbols
-LDFLAGS+=-Wl,-soname,$@,-Bsymbolic
+LDFLAGS+=-Wl,-soname,$(LIBBASE),-Bsymbolic
 endif
 endif
 
